### PR TITLE
Removed deprecated llm attribute for load_chain

### LINKED ISSUE #5224: PALChain loading fails
### System Info

langchain==0.0.176

### Who can help?

@hwchase17

### Information

- [X] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [X] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```python
from langchain.chains import PALChain
from langchain import OpenAI

llm = OpenAI(temperature=0, max_tokens=512)
pal_chain = PALChain.from_math_prompt(llm, verbose=True)
question = "Jan has three times the number of pets as Marcia. Marcia has two more pets than Cindy. If Cindy has four pets, how many total pets do the three have?"
pal_chain.save("/Users/liang.zhang/pal_chain.yaml")
loaded_chain = load_chain("/Users/liang.zhang/pal_chain.yaml")
```
Error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [17], in <cell line: 1>()
----> 1 loaded_chain = load_chain("/Users/liang.zhang/pal_chain.yaml")

File ~/miniforge3/envs/mlflow-3.8/lib/python3.8/site-packages/langchain/chains/loading.py:449, in load_chain(path, **kwargs)
    447     return hub_result
    448 else:
--> 449     return _load_chain_from_file(path, **kwargs)

File ~/miniforge3/envs/mlflow-3.8/lib/python3.8/site-packages/langchain/chains/loading.py:476, in _load_chain_from_file(file, **kwargs)
    473     config["memory"] = kwargs.pop("memory")
    475 # Load the chain from the config now.
--> 476 return load_chain_from_config(config, **kwargs)

File ~/miniforge3/envs/mlflow-3.8/lib/python3.8/site-packages/langchain/chains/loading.py:439, in load_chain_from_config(config, **kwargs)
    436     raise ValueError(f"Loading {config_type} chain not supported")
    438 chain_loader = type_to_loader_dict[config_type]
--> 439 return chain_loader(config, **kwargs)

File ~/miniforge3/envs/mlflow-3.8/lib/python3.8/site-packages/langchain/chains/loading.py:234, in _load_pal_chain(config, **kwargs)
    232 if "llm" in config:
    233     llm_config = config.pop("llm")
--> 234     llm = load_llm_from_config(llm_config)
    235 elif "llm_path" in config:
    236     llm = load_llm(config.pop("llm_path"))

File ~/miniforge3/envs/mlflow-3.8/lib/python3.8/site-packages/langchain/llms/loading.py:14, in load_llm_from_config(config)
     12 def load_llm_from_config(config: dict) -> BaseLLM:
     13     """Load LLM from Config Dict."""
---> 14     if "_type" not in config:
     15         raise ValueError("Must specify an LLM Type in config")
     16     config_type = config.pop("_type")

TypeError: argument of type 'NoneType' is not iterable
```

### Expected behavior

No errors should occur.

### PR DESCRIPTION
**Pull request by imeckr**
# Removed deprecated llm attribute for load_chain

Currently `load_chain` for some chain types expect `llm` attribute to be present but `llm` is deprecated attribute for those chains and might not be persisted during their `chain.save`.

Fixes #5224 [(issue)](https://github.com/hwchase17/langchain/issues/5224)

## Who can review?
@hwchase17
@dev2049

### CODE REVIEW

File: langchain/chains/loading.py [hunk 1]
```diff
@@ -138,19 +138,19 @@ def _load_map_reduce_documents_chain(
 
 
 def _load_llm_bash_chain(config: dict, **kwargs: Any) -> LLMBashChain:
-    if "llm" in config:
-        llm_config = config.pop("llm")
-        llm = load_llm_from_config(llm_config)
+    if "llm_chain" in config:
+        llm_chain_config = config.pop("llm_chain")
+        llm_chain = load_chain_from_config(llm_chain_config)
     elif "llm_path" in config:
```
Comment (hwchase17) [inline langchain/chains/loading.py:144]: i think llm_path should be llm_chain_path

File: langchain/chains/loading.py [hunk 2]
```diff
@@ -200,19 +200,19 @@ def _load_llm_checker_chain(config: dict, **kwargs: Any) -> LLMCheckerChain:
 
 
 def _load_llm_math_chain(config: dict, **kwargs: Any) -> LLMMathChain:
-    if "llm" in config:
-        llm_config = config.pop("llm")
-        llm = load_llm_from_config(llm_config)
+    if "llm_chain" in config:
+        llm_chain_config = config.pop("llm_chain")
+        llm_chain = load_chain_from_config(llm_chain_config)
     elif "llm_path" in config:
```
Comment (hwchase17) [inline langchain/chains/loading.py:206]: same comment here, should be llm_chain_path

File: langchain/chains/loading.py [hunk 3]
```diff
@@ -229,11 +229,11 @@ def _load_map_rerank_documents_chain(
 
 
 def _load_pal_chain(config: dict, **kwargs: Any) -> PALChain:
-    if "llm" in config:
-        llm_config = config.pop("llm")
-        llm = load_llm_from_config(llm_config)
+    if "llm_chain" in config:
+        llm_chain_config = config.pop("llm_chain")
+        llm_chain = load_chain_from_config(llm_chain_config)
     elif "llm_path" in config:
```
Comment (hwchase17) [inline langchain/chains/loading.py:235]: same comment here

**Review by hwchase17 — COMMENTED** [review]
thanks for this! one comment

File: langchain/chains/loading.py (hunk 1 above)
Comment (imeckr) [reply langchain/chains/loading.py]: fixed

File: langchain/chains/loading.py (hunk 2 above)
Comment (imeckr) [reply langchain/chains/loading.py]: done

**Comment by imeckr** [comment]
@hwchase17 thanks for the comment, I have made the changes.

**Review by hwchase17 — COMMENTED** [review]
I guess this also isn’t backwards compatible… would be nice to maintain that

**Comment by imeckr** [comment]
> I guess this also isn’t backwards compatible… would be nice to maintain that

@hwchase17 added

**Review by hwchase17 — APPROVED** [review]
thanks!

**Pull request merged**